doc/source/rllib/single-agent-episode.rst

```diff
@@ -0,0 +1,214 @@
+.. include:: /_includes/rllib/we_are_hiring.rst
+
+.. include:: /_includes/rllib/new_api_stack.rst
+
+.. include:: /_includes/rllib/new_api_stack_component.rst
+
+
+Episodes
+========
+
+RLlib stores and transports all trajectory data in the form of `Episodes`, in particular
+:py:class:`~ray.rllib.env.single_agent_episode.SingleAgentEpisode` for single-agent setups
+and :py:class:`~ray.rllib.env.multi_agent_episode.MultiAgentEpisode` for multi-agent setups.
+The data is translated from this `Episode` format to tensor batches (including a possible move to the GPU)
+only immediately before a neural network forward pass.
+
+.. figure:: images/episodes/usage_of_episodes.svg
+    :width: 750
+    :align: left
+
+    **Episodes** are the main vehicle to store and transport trajectory data across the different components
+    of RLlib (for example from `EnvRunner` to `Learner` or from `ReplayBuffer` to `Learner`).
+    One of the main design principles of RLlib's new API stack is that all trajectory data is kept in such episodic form
+    for as long as possible. Only immediately before the neural network passes, "connectors" translate lists of Episodes into tensor
```

Comment:
Ah here they come.

Reply:
Yeah, but also added this to the actual text.